src/sparsebase/io/mtx_writer.cc

```diff
@@ -0,0 +1,320 @@
+#include "sparsebase/io/mtx_writer.h"
+
+#include <string>
+
+#include "sparsebase/config.h"
+#include "sparsebase/io/sparse_file_format.h"
+#include "sparsebase/io/writer.h"
+#include "sparsebase/format/coo.h"
+#include "sparsebase/format/csr.h"
+
+namespace sparsebase::io {
+
+template <typename IDType, typename NNZType, typename ValueType>
+MTXWriter<IDType, NNZType, ValueType>::MTXWriter(
+    std::string filename, std::string object, std::string format, std::string field, std::string symmetry)
+    : filename_(filename),
+      object_(object),
+      format_(format),
+      field_(field),
+      symmetry_(symmetry) {}
+
+template <typename IDType, typename NNZType, typename ValueType>
+void MTXWriter<IDType, NNZType, ValueType>::WriteCOO(
+    format::COO<IDType, NNZType, ValueType> *coo) const {
+      
+      //illegal parameter checks
+      if (object_ != "matrix" && object_ != "vector") 
+      {
+        throw utils::ReaderException("Illegal value for the 'object' option in matrix market header");
```

Comment:
These exceptions should be to a "WriterException" rather than a "ReaderException". Could you create a new exception class for writer exceptions and use it instead for exceptions in Writer classes?

Reply:
Turns out, there is already a ReaderException class :), fixed now.